File: codeforces/gyms/100200.asc04/F.hs
{-# LANGUAGE CPP #-}
import Control.Monad
import Data.Maybe
import qualified Data.ByteString.Char8 as C

interact' :: (C.ByteString -> String) -> IO ()
#ifndef __WATASHI__
interact' = hInteract "positive.in" "positive.out"

hInteract :: FilePath -> FilePath -> (C.ByteString -> String) -> IO ()
hInteract input output f =
  liftM f (C.readFile input) >>= writeFile output
#else
interact' = interact . (. C.pack)
#endif

main :: IO ()
main = interact' $ \input ->
  let (n:a) = map readInt $ C.words $ input in
  (++"\n") $ show $ max 0 $ (n-) $ (2*) $ length $ filter (<0) $ a
  where
    readInt = fst . fromJust . C.readInt

File: codeforces/gyms/100200.asc04/F.cs
using System;
using System.IO;

public class F {
#if !__WATASHI__
  class FakeConsole: StreamWriter {
    StreamReader reader;

    public FakeConsole(string input, string output): base(output) {
      reader = new StreamReader(input);
      AutoFlush = true;
    }

    public int Read() {
      return reader.Read();
    }

    public string ReadLine() {
      return reader.ReadLine();
    }
  }

  static FakeConsole Console = new FakeConsole("positive.in", "positive.out");
#endif

  public static void Main(String[] args) {
    int n = int.Parse(Console.ReadLine());
    string s = Console.ReadLine();
    foreach (char c in s) {
      if (c == '-') {
        n -= 2;
      }
    }
    Console.WriteLine(Math.Max(n, 0));
  }
}
